### PR TITLE
Status Pull for Blake Enyart

### PR DESCRIPTION
**Pull request by blake-enyart**

### CODE REVIEW

**Comment by blake-enyart** [comment]
Blake Enyart - ActiveRecord Obstacle Course

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 1]
```diff
@@ -250,11 +253,11 @@
     ]
 
     # ----------------------- Using Ruby -------------------------
-    items = Item.all.map { |item| item unless items_not_included.include?(item) }.compact
+    # items = Item.all.map { |item| item unless items_not_included.include?(item) }.compact
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    items = Item.where.not(id: items_not_included.pluck(:id))
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: in this scenario, `pluck` isn't necessary and would actually go do a second lookup on the database. All you need here is `.where.not(id: items_not_included)`, ActiveRecord will figure out the IDs of the objects you already listed.

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 2]
```diff
@@ -312,17 +315,17 @@
...
+    #     order.items.map { |item| item.name }
+    #   end
+    # end
+    #
+    # names = names.flatten
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    names = Order.order('order_items.id').includes(:items).pluck(:name)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: shouldn't have needed the order_items.id sorting here

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 3]
```diff
@@ -483,12 +486,14 @@
...
-    order_ids = OrderItem.where(item_id: @item_4.id).map(&:order_id)
-    orders = order_ids.map { |id| Order.find(id) }
+    # order_ids = OrderItem.where(item_id: @item_4.id).map(&:order_id)
+    # orders = order_ids.map { |id| Order.find(id) }
     # -----------------------------------------------------------
 
     # ------------------ Improved Solution ----------------------
-    #  Solution goes here
+    orders = Order.joins(items: :order_items)
+                  .where('items.id': @item_4.id)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: alternate: `.where(items: {id: @item_4})`

**Pull request closed without merging**